cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -163,6 +173,11 @@ public void deleteNamespace(final Id.Namespace namespaceId)
                                                   "Some programs are currently running in namespace " + namespaceId);
     }
 
+    if (areAdaptersStarted(namespaceId)) {
+      throw new NamespaceCannotBeDeletedException(namespaceId,
+                                                  "Some adapters are currently started in namespace " + namespaceId);
```

Comment:
Change error to: "Some adapters are in started state, stop them before deleting namespace"